### ISSUE TITLE
Refactor online volume resize unit tests to be table driven

### ISSUE THREAD

**Issue by msau42** [kubernetes/kubernetes#71105]
**What happened**:
These test cases are very hard to follow because they keep toggling the feature gates on and off, and subsequent test cases depend on the behavior of the previous.

@kubernetes/sig-storage-bugs 

/kind bug

**Comment by msau42** [comment]
https://sourcegraph.com/github.com/kubernetes/kubernetes/-/blob/pkg/kubelet/volumemanager/populator/desired_state_of_world_populator_test.go#L466

**Comment by avorima** [comment]
@msau42 Hey, I'd like to take this if nobody is working on it

**Comment by msau42** [comment]
Hi @avorima thanks for volunteering! Let us know if you have any questions
